modules/bitgo/src/v2/coins/sol.ts

```diff
@@ -160,11 +165,34 @@ export class Sol extends BaseCoin {
       throw new Error('Tx memo does not match with expected txParams recipient memo');
     }
     if (txParams.recipients) {
+      totalAmount['tsol'] = new BigNumber(0);
       for (const recipients of txParams.recipients) {
-        totalAmount = totalAmount.plus(recipients.amount);
+        // totalAmount based on each token
+        if (recipients.tokenName != undefined) {
+          if (!(recipients.tokenName in totalAmount)) {
+            totalAmount[recipients.tokenName] = new BigNumber(0);
+          }
+          totalAmount[recipients.tokenName] = totalAmount[recipients.tokenName].plus(recipients.amount);
+        } else {
+          totalAmount['tsol'] = totalAmount['tsol'].plus(recipients.amount);
```

Comment:
can't hard code `'tsol'` because it will be different for production, which uses `'sol'`.

use `this.getChain()` instead.